Run `yamllint` to give out "better" formatting errors.

I was running `ilab diff` and got this error:
```
ERROR 2024-04-01 11:49:37,865 generate_data.py:560 error found in file taxonomy/compositional_skills/writing/freeform/jokes/achievements/qna.yaml: line too long (126 > 120 characters) 2 line-length
```

This errored out and didn't actually tell me where the error(s) is/was.

Running a `yamllint` against it:
```bash
(venv) [gpu@ava src]$ yamllint -c yamllint_config.yaml taxonomy/compositional_skills/writing/freeform/jokes/achievements/qna.yaml
taxonomy/compositional_skills/writing/freeform/jokes/achievements/qna.yaml
  2:121     error    line too long (126 > 120 characters)  (line-length)
  90:121    error    line too long (123 > 120 characters)  (line-length)
  139:121   error    line too long (205 > 120 characters)  (line-length)
  141:121   error    line too long (153 > 120 characters)  (line-length)
  143:121   error    line too long (310 > 120 characters)  (line-length)
  149:121   error    line too long (359 > 120 characters)  (line-length)
  151:121   error    line too long (215 > 120 characters)  (line-length)
  155:121   error    line too long (192 > 120 characters)  (line-length)
  157:121   error    line too long (316 > 120 characters)  (line-length)
  165:121   error    line too long (160 > 120 characters)  (line-length)
  173:121   error    line too long (223 > 120 characters)  (line-length)
  183:121   error    line too long (132 > 120 characters)  (line-length)
  190:121   error    line too long (156 > 120 characters)  (line-length)
  196:121   error    line too long (400 > 120 characters)  (line-length)
```

Shows a significant difference with a location of the actual error(s).

The `yamillint_conf.yaml` I created this:
```yaml
---
rules:
  line-length:
    max: 120
    allow-non-breakable-words: true
    allow-non-breakable-inline-mappings: false
```

This isn't _everything_ for our expectations, so my proposal is to create a "standard" `config` for `yamllint,` and run it directly so we get a more descriptive output of errors.

--- a/cli/generator/generate_data.py
+++ b/cli/generator/generate_data.py
@@ -566,11 +566,15 @@ def read_taxonomy_file(logger, file_path, yaml_rules: Optional[str] = None):
                     yaml_config = YamlLintConfig(DEFAULT_YAML_RULES)
             else:
                 yaml_config = YamlLintConfig(DEFAULT_YAML_RULES)
-            for p in linter.run(file, yaml_config):
-                errors += 1
-                logger.error(
-                    f"error found in file {file.name}: {p.desc} {p.line} {p.rule}"
-                )
+            linter_problems = list(linter.run(file, yaml_config))
+            if linter_problems:
+                lint_messages = [f"Problems found in file {file.name}"]
+                for p in linter_problems:
+                    errors += 1
+                    loc = f"{p.line}:{p.column}"
+                    desc = f"{p.desc} ({p.rule})" if p.rule else p.desc
+                    lint_messages.append(f"  {loc:<9} {p.level:<9} {desc}")
+                logger.error("\n".join(lint_messages))
                 return None, warnings, errors
         # do more explict checking of file contents
         with open(file_path, "r", encoding="utf-8") as file:

